File: sdk-android-sample/src/main/java/com/xyfindables/sdk/sample/XYBLEStatsView.java
package com.xyfindables.sdk.sample;

import android.content.Context;
import android.content.Intent;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.xyfindables.sdk.XYDevice;
import com.xyfindables.sdk.XYSmartScan;

/**
 * Created by arietrouw on 12/28/17.
 */

public class XYBLEStatsView extends LinearLayout {
    private static String TAG = XYBLEStatsView.class.getSimpleName();

    public XYBLEStatsView(final Context context, AttributeSet attrs) {
        super(context, attrs);
        XYSmartScan.instance.addListener(TAG, new XYDevice.Listener() {
            @Override
            public void entered(XYDevice device) {
                update();
            }

            @Override
            public void exited(XYDevice device) {
                update();
            }

            @Override
            public void detected(XYDevice device) {
                update();
            }

            @Override
            public void buttonPressed(XYDevice device, XYDevice.ButtonType buttonType) {

            }

            @Override
            public void buttonRecentlyPressed(XYDevice device, XYDevice.ButtonType buttonType) {

            }

            @Override
            public void statusChanged(XYSmartScan.Status status) {

            }

            @Override
            public void updated(XYDevice device) {
                update();
            }

            @Override
            public void connectionStateChanged(XYDevice device, int newState) {

            }

            @Override
            public void readRemoteRssi(XYDevice device, int rssi) {

            }

        });
    }

    public void update() {
        post(new Runnable() {
            public void run() {
                TextView pulseCountView = findViewById(R.id.pulses);
                if (pulseCountView != null) {
                    pulseCountView.setText(String.valueOf(XYSmartScan.instance.getPulseCount()));
                }

                TextView pulsePerSecView = findViewById(R.id.pulsesPerSecond);
                if (pulsePerSecView != null) {
                    pulsePerSecView.setText(String.valueOf(XYSmartScan.instance.getPulsesPerSecond()));
                }
            }
        });
    }
}

File: sdk-android-sample/src/main/java/com/xyfindables/sdk/sample/XYBLEStatsView.kt
package com.xyfindables.sdk.sample

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import android.widget.TextView
import com.xyfindables.sdk.devices.XYBluetoothDevice

import com.xyfindables.sdk.scanner.XYFilteredSmartScan

/**
 * Created by arietrouw on 12/28/17.
 */

class XYBLEStatsView(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {

    private val scanner : XYFilteredSmartScan

    val smartScanListener = object : XYFilteredSmartScan.Listener {
        override fun entered(device: XYBluetoothDevice) {
            update()
        }

        override fun exited(device: XYBluetoothDevice) {
            update()
        }

        override fun detected(device: XYBluetoothDevice) {
            update()
        }

        override fun connectionStateChanged(device: XYBluetoothDevice, newState: Int) {

        }

        override fun statusChanged(status: XYFilteredSmartScan.BluetoothStatus) {
            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        }
    }

    init {
        scanner = (context.applicationContext as XYApplication).scanner
        scanner.addListener(TAG, smartScanListener)
    }

    fun update() {
        post {
            val uptimeView = findViewById<TextView>(R.id.uptime)!!
            uptimeView.text = ("%.2f").format(scanner.uptimeSeconds)

            val pulseCountView = findViewById<TextView>(R.id.pulses)!!
            pulseCountView.text = scanner.scanResultCount.toString()

            val pulsePerSecView = findViewById<TextView>(R.id.pulsesPerSecond)!!
            pulsePerSecView.text = ("%.2f").format(scanner.resultsPerSecond)

            val devices = findViewById<TextView>(R.id.devices)
            devices.text = scanner.devices.size.toString()
        }
    }

    companion object {
        private val TAG = XYBLEStatsView::class.java.simpleName
    }
}
